man/adoc/bpftrace.adoc

```diff
@@ -0,0 +1,1500 @@
+
+*-V, --version*::
+  Print bpftrace version information
+
+*-v*::
+  verbose messages
+
+*-d*::
+  debug mode
+
+*-dd*::
+  verbose debug mode
+
+== Environment Variables
+
+Some behaviour can only be controlled through environment variables.
+This section lists all those variables.
+
+=== *BPFTRACE_STRLEN*
+
+Default: 64
+
+Number of bytes allocated on the BPF stack for the string returned by str().
+
+Make this larger if you wish to read bigger strings with str().
+
+Beware that the BPF stack is small (512 bytes), and that you pay the toll again inside printf() (whilst
+it composes a perf event output buffer).
+So in practice you can only grow this to about 200 bytes.
+
+Support for even larger strings is [being discussed](https://github.com/iovisor/bpftrace/issues/305).
+
+=== *BPFTRACE_NO_CPP_DEMANGLE*
+
+Default: 0
+
+C++ symbol demangling in userspace stack traces is enabled by default.
+
+This feature can be turned off by setting the value of this environment variable to `1`.
+
+=== *BPFTRACE_MAP_KEYS_MAX*
+
+Default: 4096
+
+This is the maximum number of keys that can be stored in a map.
+Increasing the value will consume more memory and increase startup times.
+There are some cases where you will want to: for example, sampling stack traces, recording timestamps for each page, etc.
+
+=== *BPFTRACE_MAX_PROBES*
+
+Default: 512
+
+This is the maximum number of probes that bpftrace can attach to.
+Increasing the value will consume more memory, increase startup times and can incur high performance overhead or even freeze or crash the system.
+
+=== *BPFTRACE_CACHE_USER_SYMBOLS*
+
+Default: 0 if ASLR is enabled on system and `-c` option is not given; otherwise 1
+
+By default, bpftrace caches the results of symbols resolutions only when ASLR (Address Space Layout Randomization) is disabled.
+This is because the symbol addresses change with each execution with ASLR.
+However, disabling caching may incur some performance.
+Set this env variable to 1 to force bpftrace to cache.
+This is fine if only trace one program execution.
+
+=== *BPFTRACE_VMLINUX*
+
+Default: None
+
+This specifies the vmlinux path used for kernel symbol resolution when attaching kprobe to offset.
+If this value is not given, bpftrace searches vmlinux from pre defined locations.
+See src/attached_probe.cpp:find_vmlinux() for details.
+
+=== *BPFTRACE_BTF*
+
+Default: None
+
+The path to a BTF file.
+By default, bpftrace searches several locations to find a BTF file.
+See src/btf.cpp for the details.
+
+=== *BPFTRACE_PERF_RB_PAGES*
+
+Default: 64
+
+Number of pages to allocate per CPU for perf ring buffer.
+The value must be a power of 2.
+
+If you're getting a lot of dropped events bpftrace may not be processing events in the ring buffer fast enough.
+It may be useful to bump the value higher so more events can be queued up.
+The tradeoff is that bpftrace will use more memory.
+
+== Supported architectures
+
+x86, arm64 and s390x
+
+//TODO: is this true?
```

Comment:
I think we have support for other arch's too. But unsure if it's tested much. Could list some tiers of support here. Maybe x86-64, arm64, and s390x best supported, everything may or may not be well tested, but we'll take bug fixes.